lib/board.rb

```diff
@@ -0,0 +1,45 @@
+class Board
+attr_accessor :cells
+  def initialize
+    @cells = []
+    reset!
+  end
+
+  def reset!
+    @cells.clear
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def position(position)
+    @cells[position.to_i - 1]
+  end
+
+  def full?
+    @cells.include?(" " || "") == false
+  end
+
+  def turn_count
+    @cells.select {|i| i == "X" || i == "O"}.length
+  end
+
+  def taken?(position)
+    position(position) == "X" || position(position) == "O"
+  end
+
+  def valid_move?(position)
+    position.to_i < 10 && position.to_i > 0 && taken?(position) == false
```

Comment:
You can use the `.between?` method and the negation operation in this method here. 

`position.to_i.between?(1, 9) && !taken?(position)`